### PR TITLE
docs: add java vulnerability page

### PR DESCRIPTION
**Pull request by DmitriyLewen**
## Description
Add java vulnerability page

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [x] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: docs/docs/vulnerability/languages/java.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+|------------------|:------------------------:|:-----------------|
+| jar/war/par/ear  |            ✓             | Include          |
+| pom.xml          |            ✓             | Exclude          |
+| *gradle.lockfile |            -             | Exclude          |
+
+### Jar
+To find information about your jar[^2] file, Trivy parses `pom.properties` and `MANIFEST.MF` files in your jar[^2] file and takes `required properties`[^3].
+
+If those files don't exist or don't contain enough information - Trivy uses [maven repository REST API](https://central.sonatype.org/search/rest-api-guide) to try to get missing information.
+
```
Comment (AnaisUrlichs) [inline docs/docs/vulnerability/languages/java.md:18]: could you please make this small change (my OCD would be very happy 😄) "to try to get the missing information."

File: docs/docs/vulnerability/languages/java.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+!!! Note
+    Trivy only takes information about package. We don't take list of vulnerability for package from `maven repository`(Information about data sources for java you can see [here](../detection/data-source.md))
```
Comment (AnaisUrlichs) [inline docs/docs/vulnerability/languages/java.md:20]: "Trivy only takes information about packages. We don't take a list of vulnerabilities for packages from the `maven repository`(Information about data sources for java you can see"

File: docs/docs/vulnerability/languages/java.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+
+You can disable use of `maven repository REST API`. Use `--offline-scan` flag for this. `--offline-scan` flag doesn't affect Trivy-DB.
```
Comment (AnaisUrlichs) [inline docs/docs/vulnerability/languages/java.md:23]: "disable the use"

File: docs/docs/vulnerability/languages/java.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+!!! Warning
+    Trivy may skip some dependencies(that were not found in `pom.properties` or `MANIFEST.MF`) when the `--offline-scan` flag is used.
+
+Base jar[^2] may contain inner jars[^2] within itself. To find information aobut these jars[^2] same logic is used as for base jar[^2].
```
Comment (AnaisUrlichs) [inline docs/docs/vulnerability/languages/java.md:28]: Some small changes: "To find information about these jars[^2], the same logic is used as for base jar[^2]."

File: docs/docs/vulnerability/languages/java.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+### Pom.xml
+Trivy parses your `pom.xml` file and try to find files with dependencies on your local PC(project folder[^4], relativePath field[^5] and local repository folder[^6]).
```
Comment (AnaisUrlichs) [inline docs/docs/vulnerability/languages/java.md:31]: "file and tries"

File: docs/docs/vulnerability/languages/java.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+If your local PC doesn't have necessary files - Trivy try to find info about these dependencies in [maven repository](https://repo.maven.apache.org/maven2/).
```
Comment (AnaisUrlichs) [inline docs/docs/vulnerability/languages/java.md:33]: "doesn't have the necessary files - Trivy tries to find the information about these dependencies in the [maven repository](https://repo.maven.apache.org/maven2/)."

File: docs/docs/vulnerability/languages/java.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+!!! Note
```
Comment (AnaisUrlichs) [inline docs/docs/vulnerability/languages/java.md:35]: I think this is the same comment as earlier -- why does it need to be repeated?

File: docs/docs/vulnerability/languages/java.md [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+    Trivy only takes information about package. We don't take list of vulnerability for package from `maven repository`(Information about data sources for java you can see [here](../detection/data-source.md))
+
+You can disable connection to maven repository. Use `--offline-scan` flag for this. `--offline-scan` flag doesn't affect Trivy-DB.
```
Comment (AnaisUrlichs) [inline docs/docs/vulnerability/languages/java.md:38]: You can disable connection to the maven repository by using `--offline-scan` flag. The `--offline-scan` flag does not affect the Trivy database.

**Review by AnaisUrlichs — CHANGES_REQUESTED** [review]
I left some comments :)

File: docs/docs/vulnerability/languages/java.md (hunk 1 above)
Comment (DmitriyLewen) [reply docs/docs/vulnerability/languages/java.md:18]: Done

File: docs/docs/vulnerability/languages/java.md (hunk 2 above)
Comment (DmitriyLewen) [reply docs/docs/vulnerability/languages/java.md]: Done

File: docs/docs/vulnerability/languages/java.md (hunk 3 above)
Comment (DmitriyLewen) [reply docs/docs/vulnerability/languages/java.md]: Done

File: docs/docs/vulnerability/languages/java.md (hunk 4 above)
Comment (DmitriyLewen) [reply docs/docs/vulnerability/languages/java.md]: Done

File: docs/docs/vulnerability/languages/java.md (hunk 5 above)
Comment (DmitriyLewen) [reply docs/docs/vulnerability/languages/java.md]: Done

File: docs/docs/vulnerability/languages/java.md (hunk 6 above)
Comment (DmitriyLewen) [reply docs/docs/vulnerability/languages/java.md]: Done

File: docs/docs/vulnerability/languages/java.md (hunk 7 above)
Comment (DmitriyLewen) [reply docs/docs/vulnerability/languages/java.md:35]: Some users only work with `pom.xml` files.
And it is possible that user will  check only the `pom.xml` part of document and not check `jar` part.
Then he will miss this important information or will think that this information is used only for `jars`.

File: docs/docs/vulnerability/languages/java.md (hunk 8 above)
Comment (DmitriyLewen) [reply docs/docs/vulnerability/languages/java.md]: Done

**Comment by DmitriyLewen** [comment]
@AnaisUrlichs thanks a lot for your notes!
I made changes and answered your question (https://github.com/aquasecurity/trivy/pull/3429#discussion_r1083429649)

File: docs/docs/vulnerability/languages/java.md [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+| pom.xml          |            ✓             | Exclude          |
+| *gradle.lockfile |            -             | Exclude          |
+
+### Jar
+To find information about your jar[^2] file, Trivy parses `pom.properties` and `MANIFEST.MF` files in your jar[^2] file and takes `required properties`[^3].
+
+    If those files don't exist or don't contain enough information - Trivy uses [maven repository REST API](https://central.sonatype.org/search/rest-api-guide) to try to get the missing information.
+
+!!! Note
+    Trivy only takes information about packages. We don't take a list of vulnerabilities for packages from the `maven repository`(Information about data sources for java you can see [here](../detection/data-source.md))
```
Comment (victornoel) [inline docs/docs/vulnerability/languages/java.md:20]: it looks like a space is missing between `maven repository` and (Information :)

Comment (victornoel) [reply docs/docs/vulnerability/languages/java.md:20]: actually there are a few other places where a space is missing in front of an opening parenthesis in the file :)

Comment (DmitriyLewen) [reply docs/docs/vulnerability/languages/java.md]: Thanks!
Fixed.

**Review by AnaisUrlichs — APPROVED** [review]

**Review by knqyf263 — APPROVED** [review]

**Pull request merged**